editoast/src/views/infra/mod.rs

```diff
@@ -141,6 +142,59 @@ async fn create(
     .await
 }
 
+/// Duplicate an infra
+#[post("/<infra>/clone")]
+async fn clone(infra: i64, conn: DBConnection) -> ApiResult<Custom<Json<Infra>>> {
+    conn.run(move |conn| {
+        let infra_ = Infra::retrieve(conn, infra)?;
+        let cloned_infra = Infra::create(infra_.name, conn)?;
+        let model_list = [
+            "osrd_infra_tracksectionmodel",
+            "osrd_infra_signalmodel",
+            "osrd_infra_speedsectionmodel",
+            "osrd_infra_tracksectionlinkmodel",
+            "osrd_infra_switchmodel",
+            "osrd_infra_switchtypemodel",
+            "osrd_infra_detectormodel",
+            "osrd_infra_bufferstopmodel",
+            "osrd_infra_routemodel",
+            "osrd_infra_operationalpointmodel",
+            "osrd_infra_catenarymodel",
+        ];
+        let generated_list = [
+            "osrd_infra_tracksectionlayer",
+            "osrd_infra_signallayer",
+            "osrd_infra_speedsectionlayer",
+            "osrd_infra_tracksectionlinklayer",
+            "osrd_infra_switchlayer",
+            "osrd_infra_detectorlayer",
+            "osrd_infra_bufferstoplayer",
+            "osrd_infra_operationalpointlayer",        
+            "osrd_infra_catenarylayer",];
```

Comment:
I think information could be retrieved from the config file
```suggestion
async fn clone(infra: i64, conn: DBConnection, map_layers: &State<MapLayers>) -> ApiResult<Custom<Json<Infra>>> {
    conn.run(move |conn| {
        let infra_ = Infra::retrieve(conn, infra)?;
        let cloned_infra = Infra::create(infra_.name, conn)?;
        let layer_table_names = map_layers.layers.map(|_, layer| layer.table_name).collect();
        let model_table_names = layers_table_name.map(|table_name| {table_name.replace("layer", "model")}).collect();
        model_table_names.push("osrd_infra_routemodel");
        model_table_names.push("osrd_infra_switchtypemodel");
```

Reply:
Thanks for your comment !

Reply:
For the `model_list` you can iter over the `ObjectType`.